### DISCUSSION TITLE
CompatibilityPlugin and Shared dependencies, MF sync

### DISCUSSION THREAD

**Discussion by crodriguez-plitzi** [webpack/webpack#15256] [Q&A]
Still not 100% if this should be considered a bug, so before create an issue ticket i want to confirm it

after debug for a few days i found something that i still thinking is wrong

in webpack when we import a library that its already a bundle in webpack, during the build process we call the plugin `CompatibilityPlugin` that basically transforms `__webpack_require__` to `__nested_webpack_require_${pattern.range[0]}__`

my concern is that shared dependencies that were loaded by the main bundle are under `__webpack_require__` and these should be exposed to the whole application if these are `singleton`

but testing the bundle compiled i can see things like this
![Screenshot from 2022-01-26 09-12-56](https://user-images.githubusercontent.com/86263126/151135245-3ed69456-c0aa-4124-ad21-5775a4c5ad41.png)
![Screenshot from 2022-01-26 09-14-08](https://user-images.githubusercontent.com/86263126/151135404-18432975-30ff-4446-aa23-44443359202a.png)

if i marked that `react` should be singleton, why is trying to load it from the nested webpack bundle instance `__nested_webpack_require_${pattern.range[0]}__` and this is why is loading the "duplicated react" instead the original from the bundle

i can understand that each nested dependency in the bundle are completely isolated, but for scenarios where dependencies are shared/singleton doesnt make sense

should this be considered a bug ?

i did some tests doing manual changes in the bundle related to this, and was able to make works react instance

![Screenshot from 2022-01-26 10-11-55](https://user-images.githubusercontent.com/86263126/151144545-4c353d5a-f976-4c9a-adab-6a481c45b33f.png)

from
`var versions = scope[name] = scope[name] || {};`
to
`var versions = scope[dependencyName] = __webpack_require__.S[name][dependencyName] || scope[dependencyName] || {};` (keep on mind that this is for the scenario where `__webpack_require__` is nested and we need access to the root)

doing just 1 change i was able to make it works

**Comment by alexander-akait** [comment]
Please use the issue template, respect time of other developers, thanks

Also please create reproducible test repo

**Reply by crodriguez-plitzi** [reply]
i used `other template` for that reason to not make it as an issue

**Comment by crodriguez-plitzi** [comment]
@alexander-akait i started this PR https://github.com/webpack/webpack/pull/15259
